news.js

```diff
@@ -0,0 +1,165 @@
+          image: newsItem.urlToImage, //image
+          description: newsItem.description, //description
+          source: newsItem.source.name, //publication
+          publishDate: newsItem.publishedAt.substring(0, 10), //date
+          storyLink: newsItem.url
+        };
+        newsData.publishDate = formatDate(newsData.publishDate);
+        newsArray.push(newsData);
+      });
+      displayNews(newsArray);
+    })
+    .catch(error => console.log(`Ruh-roh! ${error}`));
+}
+
+// Makes Dates look presentable
+function formatDate(date) {
+  let myDate = new Date(date);
+  const prettyDate = myDate.toDateString().substring(4, myDate.length);
+  myDate = prettyDate.split(" ");
+  const month = myDate[0];
+  const day = parseInt(myDate[1]);
+  const year = myDate[2];
+  let suffix = "";
+  switch (day) {
+    case 1:
+    case 21:
+    case 31:
+      suffix = "st";
+      break;
+    case 2:
+    case 22:
+      suffix = "nd";
+      break;
+    case 3:
+      suffix = "rd";
+      break;
+    default:
+      suffix = "th";
+      break;
+  }
+  return `${day}${suffix} ${month} ${year}`;
+}
+
+// Listeners Below :
+
+// Search by Category functionality -- footer
+let footerCategory = query("footer__list");
+
+// Category - Add click event listener
+footerCategory.addEventListener("click", function(event) {
+  event.preventDefault();
+  myCategory = `&category=${event.target.innerText}`;
+  url = `https://newsapi.org/v2/top-headlines?country=${myCountry}&apiKey=${myApiKey}${myCategory}`;
+  start(url);
+});
+
+// User Search functionality -- header
+const searchStr = query("search__text");
+const searchForm = query("header__search");
+
+//Listener for Search Term
+searchForm.addEventListener("submit", function(event) {
+  event.preventDefault();
+  let mySearchTerm = `everything?q=${searchStr.value}`;
+  url = `https://newsapi.org/v2/${mySearchTerm}&apiKey=${myApiKey}`;
+  start(url);
+});
+
+// Search by Category functionality -- header (repeated in footer)
+let headerCategory = query("header__list");
+
+// Category - Add click event listener
+headerCategory.addEventListener("click", function(event) {
+  event.preventDefault();
+  myCategory = `&category=${event.target.innerText}`;
+  url = `https://newsapi.org/v2/top-headlines?country=${myCountry}&apiKey=${myApiKey}${myCategory}`;
+  start(url);
+});
+
+// HEADLINE button functionality
+let home = query("main__nav");
+
+// Headline - Add click event listener
+home.addEventListener("click", function(event) {
+  event.preventDefault();
+  url = `https://newsapi.org/v2/top-headlines?country=${myCountry}&apiKey=${myApiKey}`;
+
+  start(url);
+});
+
+// Scroll to Top button functionality
+let backToTop = query("footer__backtotop"),
+  body = document.body,
+  docElem = document.documentElement,
+  offset = 100,
+  scrollPos;
+
+// Top button - Add scroll event listener to display button
+window.addEventListener("scroll", function(event) {
+  scrollPos = body.scrollTop || docElem.scrollTop;
```

Comment:
Setting a scroll listener on window can cause performance issues. Take a look at https://johnresig.com/blog/learning-from-twitter/ for an alternative approach.